conda/gateways/disk/lock.py

```diff
@@ -61,13 +70,16 @@ def __enter__(self):
                             self.fd, fcntl.LOCK_EX | fcntl.LOCK_NB, 1, LOCK_BYTE
                         )
                         break
-                    except OSError:
+                    except OSError as e:
                         if attempt > LOCK_ATTEMPTS - 2:
-                            raise
+                            raise LockError("Failed to acquire lock.") from e
```

Comment:
`as e`/`from e` is the default stack trace linking so we don't need to do that here